fe/fe-core/src/main/java/com/starrocks/analysis/ListPartitionDesc.java

```diff
@@ -0,0 +1,226 @@
+            }
+            boolean found = false;
+            for (ColumnDef columnDef : columnDefs) {
+                if (columnDef.getName().equals(partitionCol)) {
+                    if (columnDef.getType().isFloatingPointType() || columnDef.getType().isComplexType()) {
+                        throw new AnalysisException(String.format("Invalid partition column '%s': %s",
+                                columnDef.getName(), "invalid data type " + columnDef.getType()));
+                    }
+                    if (!columnDef.isKey() && columnDef.getAggregateType() != AggregateType.NONE) {
+                        throw new AnalysisException("The partition column could not be aggregated column");
+                    }
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                throw new AnalysisException("Partition column[" + partitionCol + "] does not exist in column list.");
+            }
+        }
+    }
+
+    private void analyzeMultiListPartition(Map<String, String> tableProperties) throws AnalysisException {
+        Set<String> multiListPartitionName = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+        List<List<String>> allMultiValues = Lists.newArrayList();
+        for (MultiItemListPartitionDesc desc : this.multiListPartitionDescs) {
+            if (!multiListPartitionName.add(desc.getPartitionName())) {
+                throw new AnalysisException("Duplicated partition name: " + desc.getPartitionName());
+            }
+            desc.analyze(this.partitionColNames.size(), tableProperties);
+            allMultiValues.addAll(desc.getMultiValues());
+        }
+        this.analyzeDuplicateValues(this.partitionColNames.size(), allMultiValues);
+    }
+
+    private void analyzeSingleListPartition(Map<String, String> copiedProperties) throws AnalysisException {
+        List<String> allValues = Lists.newArrayList();
+        Set<String> singListPartitionName = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+        for (SingleItemListPartitionDesc desc : this.singleListPartitionDescs) {
+            if (!singListPartitionName.add(desc.getPartitionName())) {
+                throw new AnalysisException("Duplicated partition name: " + desc.getPartitionName());
+            }
+            desc.analyze(this.partitionColNames.size(), copiedProperties);
+            allValues.addAll(desc.getValues());
+        }
+        this.analyzeDuplicateValues(allValues);
+    }
+
+    /**
+     * Check if duplicate values are found
+     * If the value of the member in the same position is equals, it is considered a duplicate
+     * @param partitionColSize the partition column size
+     * @param allMultiValues values from multi list partition
+     * @throws AnalysisException
+     */
+    private void analyzeDuplicateValues(int partitionColSize, List<List<String>> allMultiValues) throws AnalysisException {
```

Comment:
The implementation of this function confuses me. In fact, we only need to check if there are duplicate elements in allMultiValues. It would be better to write it in the following format: 
for (int i = 0; i < allMultiValues.size(); i++) {
     for (int j = i + 1; j < allMultiValues.size(); j++) {
           check equal (allMultiValues.get(i), allMultiValues.get(j))
     }
}

And show the duplicate partitionName to the user.